src/composables/useGetParticipants.js

```diff
@@ -85,17 +86,31 @@ export function useGetParticipants(isActive = ref(true), isTopBar = true) {
 		}
 
 		fetchingParticipants = true
-		debounceFastUpdateParticipants.clear()
-		debounceSlowUpdateParticipants.clear()
+
+		// Cancel the parallel request queue to not fetch twice
+		clearTimeout(debounceFastUpdateParticipantsTimeout)
+		debounceFastUpdateParticipantsTimeout = null
+		clearTimeout(debounceSlowUpdateParticipantsTimeout)
+		debounceSlowUpdateParticipantsTimeout = null
 
 		await store.dispatch('fetchParticipants', { token: token.value })
 		fetchingParticipants = false
 	}
 
-	const debounceFastUpdateParticipants = debounce(
-		cancelableGetParticipants, 3000)
-	const debounceSlowUpdateParticipants = debounce(
-		cancelableGetParticipants, 15000)
+	const debounceFastUpdateParticipants = () => {
+		if (debounceFastUpdateParticipantsTimeout) {
+			// Ignore consequent requests, but no longer than defined timeout
+			return
+		}
+		debounceFastUpdateParticipantsTimeout = setTimeout(cancelableGetParticipants, 3_000)
+	}
```

Comment:
It is not a `debounce`, it is called `throttle` 👀

Reply:
`throttle` has a leading function invoke :shrug:
On the other hand, `npm debounce` does not have maxWait time, and `vueuse useDebounceFn` does not have a cancel option

So how do I name it then?) maybe defer(Fast|Slow)Update?